### DISCUSSION TITLE
Access {{ URL.id }} or URL Parameters in Layout

### DISCUSSION THREAD

**Discussion by xnohat** [Budibase/budibase#4444] [Help] [answered]
Hi team,

I try to create new layout with some default elements will be share between many screen. I try to using {{ URL.id }} in layout, for filter data in Data Provider but it cannot, same with {{ State.variable }} also. It's there anyway to workaround to pass current "screen id" to layout elements ?

![image](https://user-images.githubusercontent.com/4613808/153704522-be075ce3-8e5b-43bf-8532-d6024d738522.png)
 
Thanks team for great platform,

**Comment by mjashanks** [comment]
Hi @xnohat - this is not possible at the moment - you don't have access to the current screen / URL. Additionally `{{ URL.id }}` will probably never make sense as URL parameters are specific to the URL

i.e. `/customers/:firstName/:surname` allows use of the `{{ URL.firstName }}` and `{{ URL.surname }}` parameters.

It would be good to allow access to the current screenName / URL though... will log that as a future enhancement

**Reply by xnohat** [reply]
Many thanks for your reply, I wait for this ability to access {{ URL.variable }} and {{ State.variable }} in Layouts

**Comment by xnohat** [comment, marked as answer]
Thanks @mjashanks for reply,
I have found a workaround method 
Use case: Layout for a series of Edit Row Screen ( /row/:id )
Use Javascript and get current URL.id of screen from current URL
script below will get id from last part of URL . Important note: js script in Budibase must access through "parent" object to access browser props 
```
return parent.window.location.href.split("/").pop();
```
example: 
```
https://budibase.domain.com/app_dev_1b898feb5dbf448fa798918257f01405#/edit-row-screen/ro_ta_906c3ea9f5ce4b218b9597bec3f77298_d49dd725397f46b4bce2937f94def537
```
id will be:
```
ro_ta_906c3ea9f5ce4b218b9597bec3f77298_d49dd725397f46b4bce2937f94def537
```
![image](https://user-images.githubusercontent.com/4613808/154861406-3514edc3-7fda-4254-abf8-20e8b0195780.png)

![image](https://user-images.githubusercontent.com/4613808/154860938-595c187c-aeb8-45c7-ae00-098768b951cf.png)

**Reply by hendrikjap** [reply]
interesting! Thanks for sharing
